### PR TITLE
Restrict semantic tests in the CI to only one thread.

### PR DESCRIPTION
**Pull request by gilbens-starkware**
A possible fix to the issue Tomer brought, the semantic tests are not the bottleneck anyway so it won't hurt, but I'm not sure that's how we want to handle it.

- - -
This change is [Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/4514)

### CODE REVIEW

**Review by orizi — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/4514)* status: all files reviewed, 1 unresolved discussion (waiting on @gilbens-starkware)

*[a discussion](https://reviewable.io/reviews/starkware-libs/cairo/4514#-NkucfYT7gSOAvwFEwqd:-NkucfYU4fF1tuBuKLnd:b-3c7r7x) (no related file):*
What is the solved issue?

___

**Review by gilbens-starkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/4514)* status: all files reviewed, 1 unresolved discussion (waiting on @orizi)

*[a discussion](https://reviewable.io/reviews/starkware-libs/cairo/4514#-NkucfYT7gSOAvwFEwqd:-NkufLcbBhsEcNLGhOV8:b-sw74lf) (no related file):*
<details><summary><i>Previously, orizi wrote…</i></summary><blockquote>

What is the solved issue?
</blockquote></details>

See in compiler-dev.
TLDR: The semantic ids in the tests are not deterministic since the db is shared and the order of the cargo tests may differ in different runs. 

___

**Review by orizi — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/4514)* status: all files reviewed, 1 unresolved discussion (waiting on @gilbens-starkware)

*[a discussion](https://reviewable.io/reviews/starkware-libs/cairo/4514#-NkucfYT7gSOAvwFEwqd:-NkvrFan1SUg6OrHSCWl:b-nh4upe) (no related file):*
<details><summary><i>Previously, gilbens-starkware (Gil Ben-Shachar) wrote…</i></summary><blockquote>

See in compiler-dev.
TLDR: The semantic ids in the tests are not deterministic since the db is shared and the order of the cargo tests may differ in different runs.
</blockquote></details>

wrong way to solve this - solution should be not printing numeric ids - and implementing DebugWithDb for relevant ids.

___
